### ISSUE TITLE
[Studio Flow] Always return 400 code at the end of the flow

### ISSUE THREAD

**Issue by kkday-kevinhorasia** [cybex-dev/twilio_voice#190]
Using any voice widget from studio flow will always give me error message [We are sorry, an application error has occur] at the end. But if I use my make-call function in studio flow, it will not give me any error.

Possible cause:
1. Free account vs Paid account? (already try both)
2. Wrong constant setting? (already check multiple times)
3. Studio flow bug? (tried with Twilio Client in TwiML App with no problem)

Assumption:
1. Some bug in twilio_voice package, since using native android and ios SDK from twilio doesn't give me that problem

Any idea or possible solution to this problem? Thanks.

**Comment by cybex-dev** [comment]
HI @kkday-kevinhorasia 

Thanks for reporting this issue.

I'm unfamiliar with Studio Flow - do you have a link to the platform I might further investigate the issue with?

**Comment by kkday-kevinhorasia** [comment]
Unfortunately due to confidential issue I can't share the link. But I will describe it more detail here

**Steps:**
1. Inside Studio Flow I use [Connect Call To] widget, as the name suggest it connects incoming call to callee.
2. Then I put Studio Flow [Webhook URL] in my TwiML App (instead of function service - make-call).
3. Lastly, in Flutter using `TwilioVoice.instance.call.place()` to make a call to my TwiML App which connect to my Studio Flow.

**Comment by cybex-dev** [comment]
@kkday-kevinhorasia sorry, I wasn't clear.

I'm unfamiliar with the Studio Flow platform, my best guess is https://flutterflow.io/ since "Studio Flow" didn't seem to show any results.

Do you have a link to the platform e.g. facebook.com. I understand, respect and do not need access to your account on the platform.

**Comment by kkday-kevinhorasia** [comment]
Sorry, I mean In Twilio Console Studio Flow.

**Comment by cybex-dev** [comment]
@kkday-kevinhorasia 

I see, thanks for reporting this issue - I'll be investigating over the weekend.

**Comment by cybex-dev** [comment]
@kkday-kevinhorasia 

If you've found a solution in the interim, that's fantastic. If not, I'd like to offer a few suggestions:

Review your TwiML application's code:
- The parameters sent from `twilio_voice`, in my experience, might be the cause (they might use lower case)
```json
{
    "To": "...",
    "From": "...",
}
```
This is a remenant from the old `twilio_voice` I've carrier over, with the intention of changing in future.
- Consider reviewing the calling endpoint e.g. `/makeCall` as this might not be found
- As a last resort, update your TwiML app with debug output to review request parameters to give further clarity on what's causing the issue. "We are sorry, an application error has occur" is usually a catch-all error message for something akin to `500/Internal Server Error` (or along those lines)

**Comment by kkday-kevinhorasia** [comment]
Thanks for the reply. I will give a simple example that give me that error:

1. In my TwiML app `request url` link to my Studio Flow.
2. Inside my Studio Flow connect incoming call to `Say` widget as in the picture below

![Greenshot 2023-11-17 09 21 11](https://github.com/cybex-dev/twilio_voice/assets/146720022/8db31a02-4414-4e57-8040-91e3f379d173)

3. From my Flutter App make call to my TwiML App (which redirect to my Studio Flow)
```
TwilioVoice.instance.call.place(
      from: caller, // My own identity
      to: callee, // any (since it will only hear message only)
      extraOptions: extraOptions,
);
```
4. I will hear the message from my Studio Flow, but I will also hear the error message "We are sorry, an application error has occur". The execution also will not stop properly and it return 400 error code as we can see from the picture below

![Greenshot 2023-11-09 09 32 10](https://github.com/cybex-dev/twilio_voice/assets/146720022/bfd8a58b-a7af-4e7a-86b6-47cb5375efcb)
![Greenshot 2023-11-09 09 34 54](https://github.com/cybex-dev/twilio_voice/assets/146720022/0ca01d38-491f-475d-acb3-0b5139ed07ce)

**Comment by cybex-dev** [comment]
> Thanks for the reply. I will give a simple example that give me that error:
> 
> 1. In my TwiML app `request url` link to my Studio Flow.
> 2. Inside my Studio Flow connect incoming call to `Say` widget as in the picture below
> 
> ![Greenshot 2023-11-17 09 21 11](https://user-images.githubusercontent.com/146720022/283641129-8db31a02-4414-4e57-8040-91e3f379d173.png)
> 
> 3. From my Flutter App make call to my TwiML App (which redirect to my Studio Flow)
> 
> ```
> TwilioVoice.instance.call.place(
>       from: caller, // My own identity
>       to: callee, // any (since it will only hear message only)
>       extraOptions: extraOptions,
> );
> ```
> 
> 4. I will hear the message from my Studio Flow, but I will also hear the error message "We are sorry, an application error has occur". The execution also will not stop properly and it return 400 error code as we can see from the picture below
> 
> ![Greenshot 2023-11-09 09 32 10](https://user-images.githubusercontent.com/146720022/283642653-bfd8a58b-a7af-4e7a-86b6-47cb5375efcb.png) ![Greenshot 2023-11-09 09 34 54](https://user-images.githubusercontent.com/146720022/283642805-0ca01d38-491f-475d-acb3-0b5139ed07ce.png)

_Might I recommend obfuscating some details_

First thoughts, `To` field is empty - does the same occur with a valid client provided?

**Comment by kkday-kevinhorasia** [comment]
Yes, with valid client provided (call is connected to callee), the `To` field at the first flow will have value, but as the flow goes to the second widget the `To` parameter will be empty. 

This behavior also occur when I am calling from Twilio Client in TwiML App, but calling from Twilio Client will not give me any error and the flow end normally.

![image](https://github.com/cybex-dev/twilio_voice/assets/146720022/41da84bc-e8a5-462a-801c-cfc3a110fd07)

**Comment by cybex-dev** [comment]
> Yes, with valid client provided (call is connected to callee), the `To` field at the first flow will have value, but as the flow goes to the second widget the `To` parameter will be empty.
> 
> This behavior also occur when I am calling from Twilio Client in TwiML App, but calling from Twilio Client will not give me any error and the flow end normally.
> 
> ![image](https://user-images.githubusercontent.com/146720022/283644405-41da84bc-e8a5-462a-801c-cfc3a110fd07.png)

I see - we're using call flows but haven't run into this problem. I'll need to investigate.

_Though, the audible message "An application error has occurred" usually indicates a TwiML (or related) issue._

**Comment by kkday-kevinhorasia** [comment]
Hi @cybex-dev,
I just wondering maybe the problem is caused by not up to date dependencies?

**Comment by kkday-kevinhorasia** [comment]
**UPDATED**

The problem is actually from the Twilio Android and iOS SDK and not this package.
It seems that Twilio team already fix it.

Twilio reference ticket: # 14617521

**Comment by cybex-dev** [comment]
Closing for now.
